src/sinks/datadog/logs/sink.rs

```diff
@@ -158,96 +126,148 @@ impl From<io::Error> for RequestBuildError {
     }
 }
 
+impl From<serde_json::Error> for RequestBuildError {
+    fn from(error: serde_json::Error) -> RequestBuildError {
+        RequestBuildError::Json { error }
+    }
+}
+
 struct LogRequestBuilder {
     default_api_key: Arc<str>,
-    encoding: JsonEncoding,
+    transformer: Transformer,
     compression: Compression,
 }
 
-impl RequestBuilder<(Option<Arc<str>>, Vec<Event>)> for LogRequestBuilder {
-    type Metadata = (Arc<str>, EventFinalizers);
-    type Events = Vec<Event>;
-    type Encoder = JsonEncoding;
-    type Payload = Bytes;
-    type Request = LogApiRequest;
-    type Error = RequestBuildError;
+struct CountingWrite<'a, T> {
+    inner: T,
+    count: &'a std::cell::Cell<usize>,
+}
+
+impl<'a, T> CountingWrite<'a, T> {
+    fn into_inner(self) -> T {
+        self.inner
+    }
+}
 
-    fn compression(&self) -> Compression {
-        self.compression
+#[allow(clippy::disallowed_methods)]
+impl<'a, T: std::io::Write> std::io::Write for CountingWrite<'a, T> {
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        self.count.set(self.count.get() + buf.len());
+        self.inner.write(buf)
     }
 
-    fn encoder(&self) -> &Self::Encoder {
-        &self.encoding
+    fn flush(&mut self) -> io::Result<()> {
+        self.inner.flush()
     }
+}
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏This seems like it's generic enough that it could live in a more common/shared area. WDYT ?

Reply:
I agree in theory, but this is the only place we're using it and I'm honestly not sure it'd be more discoverable in a util module or similar. I think if we expand this pattern to another sink, that would be a good time to put this somewhere else.